syntax = "proto3";


message Agent {
  // Public key used to sign transactions.
  // Also the unique ID.
  string public_key = 1;

  // A human readable name identifying the Agent
  string name = 2;

  int32 role = 3;

  // Unix UTC timestamp of approximately when this agent was registered
  uint64 timestamp = 4;
}

// Handle hash collisions.
message AgentContainer {
  repeated Agent entries = 1;
}